File: C#/SearchMatrix.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {
        
        List<int[]> bigList = matrix.ToList();
        
        for (int i = 0; i < matrix.GetLength(0); i++)
        {            
            if (Array.Exists(matrix[i], x => x == target))
            {
                return true;
            }
        }
        
        return false;
    }
}

File: Python/searchMatrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:

        numRows = len(matrix)
        numCols = len(matrix[0])
        row = 0
        col = 0

        while (row < numRows):
            if ((target >= matrix[row][0]) & (target <= matrix[row][numCols - 1])):
                while (col < numCols):
                    if (target == matrix[row][col]):
                        return True
                    col += 1

            elif ((target < matrix[row][0])):
                return False
            
            row += 1


        return False